src/Lucene.Net/Search/DefaultSimilarity.cs

```diff
@@ -105,4 +107,35 @@ public virtual bool DiscountOverlaps
 	        set { internalDiscountOverlaps = value; }
 	    }
 	}
+
+    /// <summary>Expert: light weight scoring implementation that disables idfExplanation. </summary>
+    [Serializable]
+    public class LightWeightSimilarity : DefaultSimilarity
+    {
+        public override Explanation.IDFExplanation IdfExplain(ICollection<Term> terms, Searcher searcher, IState state)
+        {
+            return TheExplanation.Value;
+        }
+
+        public override Explanation.IDFExplanation IdfExplain(Term term, Searcher searcher, IState state)
+        {
+            return TheExplanation.Value;
+        }
+
+        private static readonly Lazy<LightWeightSimilarity> _instance = new Lazy<LightWeightSimilarity>(() => new LightWeightSimilarity());
+        public static Similarity Instance => _instance.Value;
+
+        private static readonly Lazy<Explanation.IDFExplanation> TheExplanation = new Lazy<Explanation.IDFExplanation>(() =>
```

Comment:
Why is this lazy and not the actual value?

Reply:
Thread-safe singleton, https://csharpindepth.com/articles/singleton, is this wrong?
I have been implementing this way for years

Reply:
But we can just do that using field init, without worrying about anything else,no?

Reply:
I think that it is possible to generate multiple instances this way if you have high pressure at startup but this is unlikely to happen and it isn't crucial to have a singleton (only on startup!) in this case since the object itself is small

Reply:
https://stackoverflow.com/questions/7095/is-the-c-sharp-static-constructor-thread-safe